Connectors/Rhino/Speckle.Connectors.GrasshopperShared/Parameters/SpeckleObjectWrapper.cs

```diff
@@ -283,17 +283,122 @@ public override bool CastFrom(object source)
+    var targetType = typeof(T);
+
+    // try our "unified" conversion
+    if (TryUnifiedCast<T>(Value.GeometryBase, out var result))
     {
-      target = (T)(object)GH_Convert.ToGeometricGoo(Value.GeometryBase);
+      target = result;
       return true;
     }
 
+    // handle IGH_GeometricGoo special case
+    if (targetType == typeof(IGH_GeometricGoo))
+    {
+      target = (T)GH_Convert.ToGeometricGoo(Value.GeometryBase);
+      return true;
+    }
     return CastToModelObject(ref target);
   }
 
+  // NOTE: attempt to shorten the code, but these gh conversions are very verbose
+  private bool TryUnifiedCast<T>(GeometryBase geometry, out T result)
+  {
+    result = default!; // initialize with null-forgiving operator - result is only used when method returns true
+    object? convertedValue = null;
+    bool success = false;
+
+    var targetType = typeof(T);
+
+    if (targetType == typeof(GH_Point))
+    {
+      var point = new Point3d();
+      if (GH_Convert.ToPoint3d(geometry, ref point, GH_Conversion.Both))
+      {
+        convertedValue = new GH_Point(point);
+        success = true;
+      }
+    }
+    else if (targetType == typeof(GH_Curve))
+    {
+      Curve? curve = null;
+      if (GH_Convert.ToCurve(geometry, ref curve, GH_Conversion.Both))
+      {
+        convertedValue = new GH_Curve(curve);
+        success = true;
+      }
+    }
+    else if (targetType == typeof(GH_Line))
+    {
+      var line = new Line();
+      if (GH_Convert.ToLine(geometry, ref line, GH_Conversion.Both))
+      {
+        convertedValue = new GH_Line(line);
+        success = true;
+      }
+    }
+    else if (targetType == typeof(GH_Brep))
+    {
+      Brep? brep = null;
+      if (GH_Convert.ToBrep(geometry, ref brep, GH_Conversion.Both))
+      {
+        convertedValue = new GH_Brep(brep);
+        success = true;
+      }
+    }
+    else if (targetType == typeof(GH_Mesh))
+    {
+      Mesh? mesh = null;
+      if (GH_Convert.ToMesh(geometry, ref mesh, GH_Conversion.Both))
+      {
+        convertedValue = new GH_Mesh(mesh);
+        success = true;
+      }
+    }
+    else if (targetType == typeof(GH_Surface))
+    {
+      Surface? surface = null;
+      if (GH_Convert.ToSurface(geometry, ref surface, GH_Conversion.Both))
+      {
+        convertedValue = new GH_Surface(surface);
+        success = true;
+      }
+    }
+#if RHINO8_OR_GREATER
+    else if (targetType == typeof(GH_SubD))
+    {
+      SubD? subd = null;
+      if (GH_Convert.ToSubD(geometry, ref subd, GH_Conversion.Both))
+      {
+        convertedValue = new GH_SubD(subd);
+        success = true;
+      }
+    }
+    else if (targetType == typeof(GH_PointCloud))
+    {
+      PointCloud? pointCloud = null;
+      if (GH_Convert.ToPointCloud(geometry, ref pointCloud, GH_Conversion.Both))
+      {
+        convertedValue = new GH_PointCloud(pointCloud);
+        success = true;
+      }
+    }
+#endif
+
+    if (success && convertedValue != null)
```

Comment:
Also, you don't need to set success above: a null converted value would indicate false, so you can test for this case and return false before setting the result if needed

Reply:
Lol, good point. Don't know why I had that. Done in [chore: remove redundant success](https://github.com/specklesystems/speckle-sharp-connectors/pull/870/commits/08b41b0b429892245958e792955dd5373663a5a7)